modules/Bio/EnsEMBL/Compara/Utils/CopyData.pm

```diff
@@ -270,207 +263,100 @@ sub clear_copy_data_cache {
-
-  Description : Helper function that escapes some special characters.
-
-=cut
-
-sub _escape {
-    my $s = shift;
-    return '\N' unless defined $s;
-    $s =~ s/\n/\\\n/g;
-    $s =~ s/\t/\\\t/g;
-    return $s;
-}
-
-
-=head2 copy_data_in_text_mode
-
-  Description : A specialized version of copy_data() for tables that don't have
-                any binary data and can be loaded with mysqlimport.
-
-=cut
-
-sub copy_data_in_text_mode {
-    my ($from_dbc, $to_dbc, $table_name, $query, $replace, $ignore_fks, $debug) = @_;
-
-    assert_ref($from_dbc, 'Bio::EnsEMBL::DBSQL::DBConnection', 'from_dbc');
-    assert_ref($to_dbc, 'Bio::EnsEMBL::DBSQL::DBConnection', 'to_dbc');
-
-    $from_dbc->reconnect() unless $from_dbc->db_handle->ping;
-    $to_dbc->reconnect() unless $to_dbc->db_handle->ping;
-
-    my $user = $to_dbc->username;
-    my $pass = $to_dbc->password;
-    my $host = $to_dbc->host;
-    my $port = $to_dbc->port;
-    my $dbname = $to_dbc->dbname;
 
-    print "Query: $query\n" if $debug;
-    my $sth = $from_dbc->prepare($query, { 'mysql_use_result' => 1 });
-    $sth->execute();
-    my $curr_row;
-
-    my $total_rows = 0;
-    do {
-        my ($fh, $filename) = tempfile("${table_name}.XXXXXX", TMPDIR => 1, UNLINK => 0);
-        my $nrows = 0;
-        my $file_size = 0;
-        # The order of the condition is important: we don't want to discard a row
-        while (($file_size < MAX_FILE_SIZE_FOR_MYSQLIMPORT) and ($curr_row = $sth->fetchrow_arrayref)) {
-            my $row = join("\t", map {_escape($_)} @$curr_row) . "\n";
-            print $fh $row;
-            $file_size += length($row);
-            $nrows++;
-            $to_dbc->disconnect_if_idle if $nrows >= MAX_ROWS_WHILE_CONNECTED;
-        }
-        close($fh);
-        unless ($curr_row) {
-            $sth->finish;
-            $from_dbc->disconnect_if_idle if $nrows >= MAX_ROWS_WHILE_CONNECTED;
-        }
-        if ($nrows) {
-            my @cmd = ('mysqlimport', "-h$host", "-P$port", "-u$user", $pass ? ("-p$pass") : (), '--local', '--lock-tables', $replace ? '--replace' : '--ignore', $dbname, $filename);
-            Bio::EnsEMBL::Compara::Utils::RunCommand->new_and_exec(\@cmd, { die_on_failure => 1, debug => $debug });
-            print "Inserted $nrows rows in $table_name\n" if $debug;
-            $total_rows += $nrows;
+    # Get table's engine to optimise the copy process
+    my ($table_engine, $init_command);
+    unless ($skip_disable_vars) {
+        $table_engine = $to_dbc->db_handle->selectrow_hashref("SHOW TABLE STATUS WHERE Name = '$table_name'")->{Engine};
+        # Speed up writing of data by disabling certain variables, write the data, then enable them back
+        print "DISABLE VARIABLES\n" if $debug;
+        if ($table_engine eq 'MyISAM') {
+            $to_dbc->do("ALTER TABLE `$table_name` DISABLE KEYS");
+        } else {
+            $init_command = "SET AUTOCOMMIT = 0; SET FOREIGN_KEY_CHECKS = 0; SET UNIQUE_CHECKS = 0;";
```

Comment:
In https://support.tigertech.net/mysql-large-inserts they say we need a `COMMIT` after inserting the data. It's not needed here because `LOAD DATA` automatically does it ?

Reply:
Must confess I did miss this bit, but it was working. I have found the reason [here](https://dev.mysql.com/doc/refman/5.6/en/implicit-commit.html):
> `SET autocommit = 1` (if the value is not already 1) implicitly ends any transaction active in the current session, as if you had done a COMMIT before executing the statement.